### DISCUSSION TITLE
I need a user interface necessarily?

### DISCUSSION THREAD

**Discussion by skhrlx** [dretax/GarHal_CSGO#106] [Q&A]
I was watching this video https://youtu.be/N2zIV4DqLM8 and my mind came with questions about cheating using kernel mode (sry if is dumb questions)

1. Necessarily, kernel cheats need to have a user interface to the cheat work? It's possible to create a cheat using only a driver whithout a user interface, and this, is more hardest to code?

2. How i can use a kernel cheat in LAN like a "pro players" ? 

(I'm learning c++ and cheating for hobbie, I dont even play csgo or other games which "you need to cheat"...)

**Comment by dretax** [comment]
@skhrlx It's more complex than that, and there are lot more things behind this "story".

1. Not really. You can write your full cheat in your driver, and execute It as-is. It's just not ideal. The most hassle is to implement a some sort of graphical rendering in kernel, that would probably use directx, and renders everything accordingly. This is one of the reasons why people don't do It, and you can probably cause further traces to detect during your development if you don't cover them accordingly, plus you can also de-stabilize kernel if you are not careful enough, and making your PC run like a toaster. I'd say this is the simplest and shortest answer I could give to this thought, but as I said there is a lot more behind this, do your research. 
2. Not entirely sure what you mean here? If you are thinking about loading and executing your cheat on official tournaments like somebody did using the steam could, you would have to be creative, research, and reverse to find an "ideal" solution to this "problem".
